tests/suites/os/tests/secureboot/index.js

```diff
@@ -42,25 +53,64 @@ class secureBoot {
 	}
 
 	async testFullDiskEncryption() {
-			Promise.all(
-				[
-					{ pattern: '/^\\/mnt\\/boot$/', name: 'Boot partition' },
-					{ pattern: '/^\\/mnt\\/sysroot\\/active$/', name: 'Root partition' },
-					{ pattern: '/^\\/mnt\\/state$/', name: 'State partition' },
-					{ pattern: '/^\\/mnt\\/data$/', name: 'Data partition' },
-				].map(args => this.test.resolveMatch(
-						this.worker.executeCommandInHostOS(
-							[
-								'lsblk', '-nlo', 'MOUNTPOINT,TYPE',
-								'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
-							],
-							this.link,
-						),
-						/crypt/,
-						`${args.name} is encrypted`,
-					)
+		await Promise.all(
+			[
+				{ pattern: '/^\\/mnt\\/boot$/', name: 'Boot partition' },
+				{ pattern: '/^\\/mnt\\/sysroot\\/active$/', name: 'Root partition' },
+				{ pattern: '/^\\/mnt\\/state$/', name: 'State partition' },
+				{ pattern: '/^\\/mnt\\/data$/', name: 'Data partition' },
+			].map(args => this.test.resolveMatch(
+					this.worker.executeCommandInHostOS(
+						[
+							'lsblk', '-nlo', 'MOUNTPOINT,TYPE',
+							'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
+						],
+						this.link,
+					),
+					/crypt/,
+					`${args.name} is encrypted`,
 				)
-			);
+			)
+		);
+
+		if (!this.qmp) {
+			this.test.comment('QMP is unavailable, skipping encryption tampering test');
+			return
+		}
+
+		const label = "resin-data";
+		await this.worker.executeCommandInHostOS(
+			['while systemctl is-active balena balena-supervisor systemd-journald;',
+					 'do systemctl stop balena balena-supervisor systemd-journald; done',
+				 `&& while umount /dev/disk/by-label/${label}; do : ; done`,
+				 `&& cryptsetup luksClose /dev/disk/by-label/${label}`,
+				 `&& mkfs.ext4 -L ${label} -F /dev/disk/by-partlabel/${label}`,
+			],
+			this.link,
+		).then(() =>
+			this.test.resolves(() => {
+				var waitForFailedBootPromise = new Promise((resolve, reject) => {
```

Comment:
This part can probably be generalized to work for both QEMU/non-QEMU devices.

Reply:
+1 for factoring this out, it would be very useful if we could just `waitForRebootLoop()` from other tests without the need to repeat this.